### PR TITLE
[Service Fabric] Add pfx and output

### PR DESCRIPTION
**Pull request by QingChenmsft**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [ ] Each command and parameter has a meaningful description.
- [ ] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

File: src/command_modules/azure-cli-servicefabric/azure/cli/command_modules/servicefabric/custom.py [hunk 1]
```diff
@@ -1452,8 +1452,20 @@ def _download_secret(vault_base_url, secret_name, file_path, encoding=None, secr
                         crypto.FILETYPE_PEM, cert))
             f_pem.close()
         except Exception as ex:  # pylint: disable=broad-except
-            if os.path.isfile(file_path):
-                os.remove(file_path)
+            if os.path.isfile(pem_path):
+                os.remove(pem_path)
+            raise ex
+
+    if pfx_path is not None:
```
Comment (troydai) [inline src/command_modules/azure-cli-servicefabric/azure/cli/command_modules/servicefabric/custom.py]: Why not simply:
```
if pfx_path:
```

File: src/command_modules/azure-cli-servicefabric/azure/cli/command_modules/servicefabric/custom.py [hunk 2]
```diff
@@ -1452,8 +1452,20 @@ def _download_secret(vault_base_url, secret_name, file_path, encoding=None, secr
...
+        try:
+            import base64
+            decoded = base64.b64decode(secret_value)
+            p12 = crypto.load_pkcs12(decoded)
+            with open(pfx_path, 'wb') as f:
+                f.write(decoded)
+        except Exception as ex:  # pylint: disable=broad-except
```
Comment (troydai) [inline src/command_modules/azure-cli-servicefabric/azure/cli/command_modules/servicefabric/custom.py]: Why do you disable the pylint rule here? Can you narrow the exception types are expected here?

File: src/command_modules/azure-cli-servicefabric/azure/cli/command_modules/servicefabric/custom.py [hunk 3]
```diff
@@ -1431,16 +1431,16 @@ def import_certificate(vault_base_url, certificate_name, certificate_data,
     return result
 
 
-def _download_secret(vault_base_url, secret_name, file_path, encoding=None, secret_version=''):  # pylint: disable=unused-argument
+def _download_secret(vault_base_url, secret_name, pem_path, pfx_path, encoding=None, secret_version=''):  # pylint: disable=unused-argument
     client = _get_keyVault_not_arm_client()
     secret = client.get_secret(vault_base_url, secret_name, secret_version)
     secret_value = secret.value
-    if file_path is not None:
+    if pem_path is not None:
```
Comment (troydai) [inline src/command_modules/azure-cli-servicefabric/azure/cli/command_modules/servicefabric/custom.py]: Why not simply `if pem_path:`

File: src/command_modules/azure-cli-servicefabric/azure/cli/command_modules/servicefabric/custom.py (hunk 1 above)
Comment (QingChenmsft) [reply src/command_modules/azure-cli-servicefabric/azure/cli/command_modules/servicefabric/custom.py]: done.

File: src/command_modules/azure-cli-servicefabric/azure/cli/command_modules/servicefabric/custom.py (hunk 3 above)
Comment (QingChenmsft) [reply src/command_modules/azure-cli-servicefabric/azure/cli/command_modules/servicefabric/custom.py]: done

File: src/command_modules/azure-cli-servicefabric/azure/cli/command_modules/servicefabric/custom.py (hunk 2 above)
Comment (QingChenmsft) [reply src/command_modules/azure-cli-servicefabric/azure/cli/command_modules/servicefabric/custom.py]: this logic is from Key vault. any exception happens we don't want output the cert file. 

https://github.com/Azure/azure-cli/blob/master/src/command_modules/azure-cli-keyvault/azure/cli/command_modules/keyvault/custom.py#L554

Comment (troydai) [reply src/command_modules/azure-cli-servicefabric/azure/cli/command_modules/servicefabric/custom.py]: A similar code snippet does exist in current code base. But it doesn't necessarily make it right. We're going to revisit all the `broad-except` pylint rule disable in the future.

Comment (troydai) [reply src/command_modules/azure-cli-servicefabric/azure/cli/command_modules/servicefabric/custom.py]: Filed issue: https://github.com/Azure/azure-cli/issues/4486

**Pull request merged**